RxSwiftWorkshop/Maps/Model/GeolocatorSpec.swift

```diff
@@ -34,47 +33,89 @@ class GeolocatorSpec: QuickSpec {
                 sut = nil
             }
 
-            it("should pass geolocation request to its SystemGeolocator") {
-                guard case let (disposeBag?, systemGeolocator?) = unwrap(disposeBag, systemGeolocator) else { return }
+            condition("always") {
                 let address = "Valencia Airport"
-                sut?.geolocate(address: address).subscribe().disposed(by: disposeBag)
-                systemGeolocator.verifyCall(
-                    withIdentifier: FakeSystemGeolocator.Identifiers.geocodeAddressString, arguments: [address]
-                )
+                setup {
+                    sut?.geolocate(address: address).subscribe().disposed(by: disposeBag!)
+                }
+                it("should pass geolocation request to its SystemGeolocator") {
+                    systemGeolocator!.verifyCall(
+                        withIdentifier: FakeSystemGeolocator.Identifiers.geocodeAddressString, arguments: [address]
+                    )
+                }
             }
 
-            it("should return geolocation from its SystemGeolocator") {
-                guard case let (sut?, disposeBag?, systemGeolocator?) = unwrap(sut, disposeBag, systemGeolocator) else { return }
+            condition("when SystemGeolocator succeeds in finding location") {
                 let location = CLLocation(latitude: 123.123, longitude: 132.132)
-                systemGeolocator.fixedLocation = location
                 var result: CLLocation?
-                sut.geolocate(address: "Valencia Airport").subscribe(onNext: { result = $0 }).disposed(by: disposeBag)
-                expect(result?.coordinate).to(equal(location.coordinate))
+                setup {
+                    systemGeolocator!.fixedLocation = location
+                    sut!.geolocate(address: "Valencia Airport")
```

Comment:
I think, I made a mistake for the second time. You're right – system geolocator doesn't expose reactive API. So the `testableObservable` doesn't make much sense, which is a pity, because that would be nice.

In this case you have capture next result, indeed. I don't like it very much, but there's nothing you can actually do. 

But you might be using `RxBlocking`, which allows you to write:

```swift
let result = sut.geolocate(address: "Valencia Airport").toBlocking().first
```

It makes asynchronious more synchronious in a, not very nice, way.